### PR TITLE
#347 extend pairing on approve proposal

### PR DESCRIPTION
**Pull request by llbartekll**
extend pairing on every approve proposal on wallet

### CODE REVIEW

File: Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift [hunk 1]
```diff
@@ -82,7 +81,15 @@ final class ApproveEngine {
         let proposeResponse = SessionType.ProposeResponse(relay: relay, responderPublicKey: selfPublicKey.hexRepresentation)
         let response = JSONRPCResponse<AnyCodable>(id: payload.wcRequest.id, result: AnyCodable(proposeResponse))
 
+        guard var pairing = pairingStore.getPairing(forTopic: payload.topic) else {
+            throw Errors.pairingNotFound
+        }
+
         try await networkingInteractor.respond(topic: payload.topic, response: .response(response), tag: payload.wcRequest.responseTag)
+
+        try pairing.updateExpiry()
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift:90]: we also extending pairing in `handleProposeResponse`. So in approve method we extending for wallet, and in handleProposeResponse we extending for dapp, right?

Comment (llbartekll) [reply Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift:90]: correct

File: Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift [hunk 2]
```diff
@@ -82,7 +81,15 @@ final class ApproveEngine {
         let proposeResponse = SessionType.ProposeResponse(relay: relay, responderPublicKey: selfPublicKey.hexRepresentation)
         let response = JSONRPCResponse<AnyCodable>(id: payload.wcRequest.id, result: AnyCodable(proposeResponse))
 
+        guard var pairing = pairingStore.getPairing(forTopic: payload.topic) else {
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift:84]: do we care about pairing is active / inactive ?

Comment (llbartekll) [reply Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift:84]: I think on wallet pairing is always active

Comment (llbartekll) [reply Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift:84]: exaclty

**Review by flypaper0 — APPROVED** [review]

**Pull request merged**